src/praisonai-agents/praisonaiagents/context/optimizer.py

```
@@ -576,6 +577,222 @@ def _llm_summarize(self, messages: List[Dict[str, Any]], max_tokens: int) -> str
+            optimized_tokens = estimate_messages_tokens(result.messages)
+            
+            return result.messages, OptimizationResult(
+                original_tokens=original_tokens,
+                optimized_tokens=optimized_tokens,
+                tokens_saved=result.tokens_saved,
+                strategy_used=OptimizerStrategy.SUMMARIZE,
+                messages_removed=result.middle_compressed_count,
+                summary_added=True,
+            )
+            
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)